laurent22/joplin
# DESKTOP: Fixed (#2467) edit menu items and shortcuts not disabled in preview mode

**DESCRIPTION:**
This patch is a fix for the issue #2467 . Previously before this solution, the Edit menu items were not disabled on PREVIEW, this caused users to be able to use short cuts such as CTRL+B, CTRL+` on PREVIEW mode which is an unwanted behavior.

**SOLUTION:**
A function which simulates a switch was created for enabling and disabling these edit menu items, and this function is called conditionally depending on if the panes are in viewer or editor mode. it also takes account of when a notebook/note is empty, it maintains the behavior of disabling these edit menu items/shortcuts when a note has no content.

**HOW TO TEST**
Use the following test cases to confirm this solution
- **Create an empty notebook and click the edit button**: 
    EXPECTED RESULT:  'copy', 'paste', 'cut', 'select all', 'bold', 'italic', 'link', 'code', 'insert Date Time', 'Start External Editing', 'show Local Search' should be disabled
- **Add a note to a notebook and click the layout toggler to PREVIEW**: 
    EXPECTED RESULT:  'copy', 'paste', 'cut', 'select all', 'bold', 'italic', 'link', 'code', 'insert Date Time', 'Start External Editing', 'show Local Search' should be disabled
- **Add a note to a notebook and click the layout toggler to EDITOR**: 
    EXPECTED RESULT:  'copy', 'paste', 'cut', 'select all', 'bold', 'italic', 'link', 'code', 'insert Date Time', 'Start External Editing', 'show Local Search' should be enabled
- **Add a note to a notebook and click the layout toggler to EDITOR AND PREVIEW MODE**: 
    EXPECTED RESULT:  'copy', 'paste', 'cut', 'select all', 'bold', 'italic', 'link', 'code', 'insert Date Time', 'Start External Editing', 'show Local Search' should be enabled
